src/toil/test/jobStores/jobStoreTest.py

```diff
@@ -839,6 +840,49 @@ def _corruptJobStore(self):
         assert isinstance(self.master, AWSJobStore)  # type hinting
         self.master.filesBucket.delete()
 
+    def testSDBDomainsDeletedOnFailedJobstoreBucketCreation(self):
+        """
+        This test ensures that SDB domains bound to a jobstore are deleted if the jobstore bucket
+        failed to be created.  We simulate a failed jobstore bucket creation by using a bucket in a
+        different region with the same name.
+
+        :returns: None
+        """
+        from boto.sdb import connect_to_region
+        from boto.s3.connection import Location, S3Connection
+        aws_west_regions = {'us-west-1': Location.USWest,
+                            'us-west-2': Location.USWest2}
+        externalAWSRegion = [R for r, R in aws_west_regions.items() if r != self.awsRegion()][0]
+
+        testJobStoreUUID = str(uuid.uuid4())
+        # Create the nucket at the external region
+        s3 = S3Connection()
+        bucket = s3.create_bucket('domain-test-' + testJobStoreUUID + '--files',
+                                  location=externalAWSRegion)
+        options = Job.Runner.getDefaultOptions('aws:' + self.awsRegion() + ':domain-test-' +
+                                               testJobStoreUUID)
+        try:
+            with Toil(options) as toil:
+                pass
+        except JobStoreBucketExistsException:
+            # Catch the expected JobStoreBucketExistsException and ensure that the bound domains
+            # don't exist in SDB
+            sdb = connect_to_region(self.awsRegion())
+            next_token = None
+            allDomainNames = []
+            while True:
+                domains = sdb.get_all_domains(max_domains=100, next_token=next_token)
+                allDomainNames.extend([x.name for x in domains])
+                next_token = domains.next_token
+                if next_token is None:
+                    break
+            # Will the user ever have 0 domains?
```

Comment:
They might. The registry should exist but domain creation is eventually consistent, so you might not see it if this was the first time a job store was created in the region.